### DISCUSSION TITLE
关于[deband_bitrate]自动去色带的疑问

### DISCUSSION THREAD

**Discussion by GFDGIT** [hooke007/mpv_PlayKit#388] [mpv 通用]
我发现无论文件码率多少，都会启用自动去色带，所以检查了一下。

修改了目标码率，增加一行输出：
```
[deband_bitrate]
 profile-desc = 基于码率的自动去色带
 profile-cond = ((p.file_size/1024)*8)/p.duration<=100
 profile-restore = copy
 deband = yes
 term-playing-msg= "Size:${file-size}, Duration:${duration}"
```

问题：

1. 无论目标码率是多少，条件总是成立
2. `${file-size}`并不是总能获取到数值，很大概率是0B，`${duration}`则总是正常获取
3. 正常获取到时，格式是xxGB和`hh:mm:ss`，这两个单位能直接进行计算？
```
Size:4.185 GiB, Duration:01:52:13
```

**Comment by hooke007** [comment]
那就条件里加个 `p.file_size>0`
即
```
p.file_size>0 and ((p.file_size/1024)*8)/p.duration<=3000
```

term-playing-msg 输出的值是格式化后的，不是变量的原始值

续： https://github.com/hooke007/MPV_lazy/discussions/388#discussioncomment-7546246

**Reply by hooke007** [reply]
- 经测试是上游的bug https://github.com/mpv-player/mpv/issues/12869

更新到最新版主程序，不修改条件也可解决

**Reply by GFDGIT** [reply]
OK，谢谢解答。

**Reply by zhongfly** [reply]
最新构建(获得最新的修复，并体验最新的bug)
https://github.com/zhongfly/mpv-winbuild/releases/latest

**Reply by GFDGIT** [reply]
> 最新构建(获得最新的修复，并体验最新的bug)
> https://github.com/zhongfly/mpv-winbuild/releases/latest

手动更新mpv就违背了使用懒人包的本意了😮‍💨。
逐渐在折腾的道路上越走越远，当然折腾本身也包含了乐趣。

**Reply by zhongfly** [reply]
> 手动更新mpv

可以直接用更新脚本一键下载啊…只是更新下exe而已，有什么折腾的…
我觉得懒人包的主要意义是提供了一套可用的配置/脚本。
不过mpv总是会有新东西，也会有新bug就是了

**Reply by hooke007** [reply]
他的意思应该是要手动检查提交历史，防止参数失效与变动

**Reply by GFDGIT** [reply]
我想说的是mpv终究还是面向“极客”用户的，普通用户真的很难说使用一个播放器要先去查阅手册，直接用文本进行设置，甚至还需要一些编程基础。即便是其他带GUI的mpv前端，易用性上也比其他播放器差的。

作为一个“中间”用户，我虽然有一点能力查手册和文档，但却没有能力和精力去追踪上游的更新和bug了。

在这个方面，mpv-lazy真的很有意义，让普通人先忽略一些细节，直接解压就能使用。但不可否认，一旦需要进行设置或者出错，就会面临很多问题，有些问题明明很简单，但因为不熟悉就变得折腾了，学习成本也变得很高了。别的不说，单是上github都能难倒很多人。

大多数人是对画质、自定义、性能有所需求，才会发现和使用mpv，慢慢还是会需要自己进行设置的，就看个人怎么平衡这个需求和成本了。

**Reply by zhongfly** [reply]
由于我自己用的配置相当简单，播放环境也很简单，大部分时候更新都没遇到问题，所以我个人建议都是用最新，有问题就回滚版本…

**Reply by GFDGIT** [reply]
> 逐渐在折腾的道路上越走越远

这说的是我现在的状态。

并不是说升级一下mpv就很折腾，直接升级通常也是没有问题，在没有使用mpv-lazy的时候，我也是覆盖升级的，只不过只更新正式版，这种nightly版本是真的追不动。。。

**Comment by 3a311088** [comment]


**Reply by hooke007** [reply]
条件太长了没看。
profile 本身存在竞争条件限制，复杂需求建议写脚本用 file load 的 event 做标定。
